### PR TITLE
#14 Convert to cmake

### LINKED ISSUE #14: Home Screen - Spedometer
### Description

We need to be design a spedometer component to visualize the speed for the driver. 

### Acceptance Criteria

Component should take in the following 
- [ ] height
- [ ] width
- [ ] Value (Q Property)
- [ ] max speed (defaults to 100)
Should animate as the speed increases

### Proposed Solution

Look at the example spedometer in qt design studio. In fact i think you can make the entire thing in qt design studio

### Mocks

Screen Shot 2023-09-24 at 4 22 05 PM

epic #13

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Converted Code to CMAKE + WIP Spedometer Component

## Screenshots
Screenshot 2023-10-19 at 1 39 58 PM
Screenshot 2023-10-19 at 1 40 18 PM
Screenshot 2023-10-19 at 1 40 32 PM
Screenshot 2023-10-19 at 1 40 50 PM
Screenshot 2023-10-19 at 1 41 05 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #14  (issue #)

### CODE REVIEW

**Comment by Peyton-McKee** [comment]
Updated Screen Shots

File: NERODesign/content/App.qml [hunk 1]
```diff
@@ -5,14 +5,19 @@ import QtQuick 6.5
...
+    width: 800
+    height: 480
+    color: "black"
 
     visible: true
     title: "NERO"
 
-    home {
-        id: mainScreen
+    Spedometer {
```
Comment (RChandler234) [inline NERODesign/content/App.qml:15]: so why is there an App.qml in NERODesign and NERODevelopment. Like what is the purpose of NERODesign vs NERODevelopment is what I'm asking

File: NERODesign/content/Spedometer.qml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            context.strokeStyle = gradient;
+
+            context.lineWidth = ring.borderWidth;
+            context.beginPath();
+            context.arc((ring.width / 2), ring.height - gauge.verticalPadding + (gauge.verticalPadding / 10), ring.ringRadius, ring.startAngle, endAngle);
+            context.stroke();
+        }
+    }
+
+    Text {
```
Comment (RChandler234) [inline NERODesign/content/Spedometer.qml:88]: so is the Spedometer in NERODevelopment an exact copy of this one?

**Review by RChandler234 — CHANGES_REQUESTED** [review]
this is how the spedometer looks on my screen rn (I know that's not how it actually looks but idk how to fix it locally), not sure how I can check out the thermometer as well
image

File: README.md
[the remark this answers is not in the document]
Comment (Peyton-McKee) [reply README.md:1]: got moved, can move it back

File: NERODesign/content/App.qml (hunk 1 above)
Comment (Peyton-McKee) [reply NERODesign/content/App.qml:15]: theyre two projects, ones for qt design studio and the other one is for qt creator, you create the design in design studio then export it to creator

File: NERODesign/content/Spedometer.qml (hunk 2 above)
Comment (Peyton-McKee) [reply NERODesign/content/Spedometer.qml:88]: yes

**Pull request merged**